dev/tests/integration/testsuite/Magento/Customer/Controller/Account/EmailTemplateTest.php

```diff
@@ -0,0 +1,199 @@
+     *
+     * @magentoDataFixture Magento/Customer/_files/customer.php
+     * @magentoConfigFixture current_store customer/captcha/enable 0
+     */
+    public function testCustomerEmailAndPasswordChangeNotificationTemplateGreeting()
+    {
+        $this->loginByCustomerId(self::FIXTURE_CUSTOMER_ID);
+
+        $this->sendAccountEditRequest([
+            'email' => 'new.email@example.com',
+            'change_email' => 1,
+            'change_password' => 1,
+            'password' => 'new-Password1',
+            'password_confirmation' => 'new-Password1',
+        ]);
+
+        $this->assertRedirect($this->stringContains('customer/account/'));
+        $this->assertSessionMessages(
+            $this->equalTo(['You saved the account information.']),
+            MessageInterface::TYPE_SUCCESS
+        );
+
+        $this->assertSameGreeting(self::EXPECTED_GREETING, $this->transportBuilderMock->getSentMessage());
+    }
+
+    /**
+     * @covers Magento_Customer::view/frontend/email/change_password.html
+     *
+     * @magentoDataFixture Magento/Customer/_files/customer.php
+     * @magentoConfigFixture current_store customer/captcha/enable 0
+     */
+    public function testCustomerPasswordChangeNotificationTemplateGreeting()
+    {
+        $this->loginByCustomerId(self::FIXTURE_CUSTOMER_ID);
+
+        $this->sendAccountEditRequest([
+            'change_password' => 1,
+            'password' => 'new-Password1',
+            'password_confirmation' => 'new-Password1',
+        ]);
+
+        $this->assertRedirect($this->stringContains('customer/account/'));
+        $this->assertSessionMessages(
+            $this->equalTo(['You saved the account information.']),
+            MessageInterface::TYPE_SUCCESS
+        );
+
+        $this->assertSameGreeting(self::EXPECTED_GREETING, $this->transportBuilderMock->getSentMessage());
+    }
+
+    /**
+     * Wraps Customer Edit POST request
+     *
+     * @param array $customData
+     */
+    private function sendAccountEditRequest(array $customData): void
+    {
+        $basicData = [
+            'form_key' => $this->formKey->getFormKey(),
+            'firstname' => self::FIXTURE_CUSTOMER_FIRSTNAME,
+            'lastname' => self::FIXTURE_CUSTOMER_LASTNAME,
+            'current_password' => self::FIXTURE_CUSTOMER_PASSWORD
+        ];
+
+        $this->getRequest()->setMethod(HttpRequest::METHOD_POST)
+            ->setPostValue(array_merge($basicData, $customData));
+
+        $this->dispatch('customer/account/editPost');
+    }
+
+    /**
+     * Verifies if `<p class="greeting"/>` text contents equals the expected one.
+     *
+     * @param string $expectedGreeting
+     * @param EmailMessage $message
+     */
+    private function assertSameGreeting(string $expectedGreeting, EmailMessage $message)
+    {
+        $messageContent = $this->getMessageRawContent($message);
+        $emailDom = new \DOMDocument();
+        $emailDom->loadHTML($messageContent);
```

Comment:
We can introduce new "assert class" and put in 
https://github.com/magento/magento2/tree/2.4-develop/dev/tests/integration/framework/Magento/TestFramework
`Assert` namespace.

We achieve 2 goals:
- working with DOM will be encapsulated ;
- we can reuse the "assert" in the other testcases;

Thanks

Reply:
I'm going to create separate ticket for that point.

Reply:
> I'm going to create separate ticket for that point.

Please link this and a new ticket.

Reply:
https://github.com/magento/magento2/issues/27607